pkg/clusteragent/autoscaling/workload/controller_test.go

```diff
@@ -228,3 +235,51 @@ func TestLeaderCreateDeleteRemote(t *testing.T) {
 	f.RunControllerSync(true, "default/dpa-0")
 	assert.Len(t, f.store.GetAll(), 0)
 }
+
+func TestLeaderCreateDeleteLocalHeap(t *testing.T) {
+	testTime := time.Now()
+	f := newFixture(t, testTime)
+
+	dpaSpec := datadoghq.DatadogPodAutoscalerSpec{
+		TargetRef: autoscalingv2.CrossVersionObjectReference{
+			Kind:       "Deployment",
+			Name:       "app-0",
+			APIVersion: "apps/v1",
+		},
+		// Local owner means .Spec source of truth is K8S
+		Owner: datadoghq.DatadogPodAutoscalerLocalOwner,
```

Comment:
You also need a test on remote owner, which should have zero `CreationTimestamp` value first and then be created in Kubernetes and then get updated with Kubernetes creation timestamp